graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -0,0 +1,320 @@
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableColumn;
+import io.stargate.db.schema.ImmutableTable;
+import io.stargate.db.schema.Keyspace;
+import io.stargate.db.schema.Table;
+import io.stargate.graphql.schema.schemafirst.migration.CassandraSchemaHelper;
+import io.stargate.graphql.schema.schemafirst.util.Uuids;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO purge old entries
+public class SchemaSourceDao {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SchemaSourceDao.class);
+  public static final String KEYSPACE_NAME = "stargate_graphql";
+  public static final String TABLE_NAME = "schema_source";
+  @VisibleForTesting static final String NAMESPACE_COLUMN_NAME = "namespace";
+  @VisibleForTesting static final String VERSION_COLUMN_NAME = "version";
+  @VisibleForTesting static final String LATEST_VERSION_COLUMN_NAME = "latest_version";
+  @VisibleForTesting static final String CONTENTS_COLUMN_NAME = "contents";
+  @VisibleForTesting static final String APPLIED_COLUMN_NAME = "[applied]";
+
+  @VisibleForTesting
+  static final String DEPLOYMENT_IN_PROGRESS_COLUMN_NAME = "deployment_in_progress";
+
+  private static final int NUMBER_OF_RETAINED_SCHEMA_VERSIONS = 10;
+
+  @VisibleForTesting
+  static final Table EXPECTED_TABLE =
+      ImmutableTable.builder()
+          .keyspace(KEYSPACE_NAME)
+          .name(TABLE_NAME)
+          .addColumns(
+              ImmutableColumn.create(
+                  NAMESPACE_COLUMN_NAME, Column.Kind.PartitionKey, Column.Type.Varchar),
+              ImmutableColumn.create(
+                  VERSION_COLUMN_NAME,
+                  Column.Kind.Clustering,
+                  Column.Type.Timeuuid,
+                  Column.Order.DESC),
+              ImmutableColumn.create(
+                  CONTENTS_COLUMN_NAME, Column.Kind.Regular, Column.Type.Varchar),
+              ImmutableColumn.create(
+                  LATEST_VERSION_COLUMN_NAME, Column.Kind.Static, Column.Type.Timeuuid),
+              ImmutableColumn.create(
+                  DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, Column.Kind.Static, Column.Type.Boolean))
+          .build();
+
+  private final DataStore dataStore;
+
+  public SchemaSourceDao(DataStore dataStore) {
+    this.dataStore = dataStore;
+  }
+
+  public List<SchemaSource> getSchemaHistory(String namespace) throws Exception {
+    if (!tableExists()) {
+      return Collections.emptyList();
+    }
+    List<Row> row = dataStore.execute(schemaQuery(namespace)).get().rows();
+    if (row == null) {
+      return Collections.emptyList();
+    }
+    return row.stream().map(r -> toSchemaSource(namespace, r)).collect(Collectors.toList());
+  }
+
+  public SchemaSource getByVersion(
+      String namespace,
+      @SuppressWarnings("OptionalUsedAsFieldOrParameterType") Optional<UUID> maybeVersion)
```

Comment:
Nit: It feels odd that in a method named `getByVersion` the version is optional. Should this be named `getSchema`?

Reply:
Yes, it's more like "get a single version". I'll clean up all method names.